### ISSUE TITLE
dolt fails to insert multiple values into table when using DEFAULT values

### ISSUE THREAD

**Issue by noamross** [dolthub/dolt#3741]
When inserting multiple values into a table where some but not all are DEFAULT, dolt fails to handle the insert statement properly:  

```sql
create table test(id int default 1, name varchar(128));
insert into `test`(`id`) values (3), (2);
--- Query OK, 2 rows affected
insert into `test`(`id`) values (DEFAULT), (DEFAULT);
--- Query OK, 2 rows affected
insert into `test`(`id`) values (3), (DEFAULT);
--- plan is not resolved because of node '*plan.Values'
insert into `test`(`id`) values (DEFAULT), (3);
--- number of values does not match number of columns provided
```

This was found while debugging #3674 @fulghum

**Comment by fulghum** [comment]
Great find @noamross! Thank you! I'm wrapping up debugging another issue and this one is next in my queue.

**Comment by fulghum** [comment]
I've been debugging through this one today and narrowing in on the fix. I also found this related tracking issue in the go-mysql-server repo: https://github.com/dolthub/go-mysql-server/issues/527

**Comment by fulghum** [comment]
I figured out what was going on here yesterday and got a new implementation passing my new tests this morning. I still need to clean up the code a bit and double-check some edge cases, but I'll get this out to a PR shortly and we'll get the fix merged in asap.

**Comment by noamross** [comment]
Thanks! Curious to see how much farther the NoCoDB startup goes.

**Comment by fulghum** [comment]
The go-mysql-server in-memory implementation is working correctly in https://github.com/dolthub/go-mysql-server/pull/1104 . I've been debugging a few things with the Dolt layer this afternoon – it needed a little more logic than go-mysql-server's reference implementation.

I'm happy to plug in a new dolt dev build to our NocoDB test as soon I get these next changes working and let you know how far it gets! 🤞🏼

**Comment by fulghum** [comment]
I got the column default code working for Dolt this morning and tested against nocoDB. We get past the insertion into the `nc_plugins` table that was failing before 🎉, but hit an error on adding an index:
```
migration failed with error: alter table `nc_audit` add index ```nc_audit_index```(`db_alias`, `project_id`, `model_name`, `model_id`) - invalid index name ``nc_audit_index`` as they must match the regular expression ^[-$_0-9a-zA-Z]+$
```

The triple backticks obviously look odd, so I'll dig into how MySQL handles those. I'm guessing they are simply trimmed off, but I'll test and confirm. Seems like this will likely be a quick fix. 

Now that the insert column default tests are passing for Dolt, I'll tidy up the changes, and get them out for review today. Thanks for getting the NocoDB experiment started – this has turned up some nice fixes for Dolt!

**Comment by fulghum** [comment]
This fix is now merged into the `main` branch of Dolt. 🎉 This will go out in our next Dolt release, but don't hesitate to let us know if you need it faster and we're happy to go ahead and run a release for you. 

I went ahead and cut a new issue (https://github.com/dolthub/dolt/issues/3776) for the backticks in index name issue. I'm working on that fix and will keep you updated. 

I also cut a new issue for the floats with scale and precision issue we found last week (https://github.com/dolthub/dolt/issues/3778) that is also required to get NocoDB working with Dolt. 

Closing this one out since we've got this merged into main now and staged for the next release, as well as a couple other issues that we'll keep following up on. Thanks for reporting these Noam!

**Comment by noamross** [comment]
Thanks for following up on these! I wonder if NocoDB has a lot of edge-case syntax that originates from it trying to be fully compatible for multiple database back-ends.  So they end of with stuff that is not "The MySQL way" but the "Just tweaked enough to work with MySQL while being close to the others" way.

**Comment by fulghum** [comment]
@noamross – My pleasure! I've got all the NocoDB fixes so far merged into Dolt main now, and a new fix prototyped for the next issue I hit (https://github.com/dolthub/dolt/issues/3788). I tested out that last fix... and sure enough... NocoDB started up! 🎉 I poked around just a little bit to make sure I could create some data and things looked good. We may hit issues using different features of course, but it's great to see it up and running. As always, feel free to send us issues if you find anything. 

If you wanna check it out before everything gets released, I've got my changes to build from that patched branch pushed to my [dolt-nocodb fork](https://github.com/fulghum/dolt-nocodb).
